### PR TITLE
etcd(cdc): retry on etcd return raft stopped error to make owner more robust to pd member change

### LINKED ISSUE #6583: scale-in pd cause the cdc latency goes up too high and cannot goes down
* Scale-in the PD leader cause TiCDC owner switched.
image

* the checkpoint and resolved ts goes up for around 1 minutes, and elapsed for a long time.
image

Log shows that the etcd is unavailable because the PD which the etcd belongs to is already scaled in.
```
[2022/08/03 11:03:41.236 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-0.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.4.251:2379: connect: connection refused"]
[2022/08/03 11:03:41.240 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-1.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.9.136:2379: connect: connection refused"]
[2022/08/03 11:03:43.799 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-0.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.4.251:2379: connect: connection refused"]
[2022/08/03 11:03:43.801 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-1.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.9.136:2379: connect: connection refused"]
[2022/08/03 11:03:46.798 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-0.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.4.251:2379: connect: connection refused"]
[2022/08/03 11:03:46.800 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-1.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.9.136:2379: connect: connection refused"]
[2022/08/03 11:03:49.799 +08:00] [WARN] [server.go:257] ["etcd health check error"] [error="Get \"http://lingjin-0.jinling.tispace:2379/pd/api/v1/health\": dial tcp 10.244.4.251:2379: connect: connection refused"]
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #6583 

### What is changed and how it works?

Deploy 3 pd instances, run `TiUP cluster scale-in` one instance, which can cause the cdc owner to switch, and replication latency goes up.

This is caused by etcd return the error `raft: stopped`. see the PR for more detail.

This PR can fix the problem by retrying on the error, no more owner switch and replication latency does not goes up.

```
[2022/08/03 20:25:17.138 +08:00] [WARN] [client.go:97] ["etcd RPC failed"] [RPC=Txn] [error="rpc error: code = Unknown desc = raft: stopped"]
[2022/08/03 20:25:19.325 +08:00] [WARN] [etcd_worker.go:448] ["Etcd transaction took too long"] [duration=2.188130721s]
```
we can still see the log warning that `raft: stopped`, but this no more interrupts cdc replication task.

After this PR, TiCDC still suffer from scale-in the pd leader caused high replication latency but no owner switch, since at the moment TiUP does not transfer the leadership in advance before delete the instance from the cluster .

#### Tests 

 - Manual test (add detailed scripts or steps below)
* deploy 3 pd instances
* `tiup cluster scale-in` any one of the instances.

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -105,6 +105,8 @@ require (
 	upper.io/db.v3 v3.7.1+incompatible
 )
 
+require go.etcd.io/etcd/raft/v3 v3.5.2
```
Comment (asddongmen) [inline go.mod:108]: move this line into the require block above.

**Review by asddongmen — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**